app.js

```diff
@@ -170,55 +22,28 @@ for (let i = 2; i < args.length; i++) {
     }
 
     if (args[i] === "--output=xml") {
-        isToXML = true;
+        output = "xml";
     }
 }
 
-
-if (!isToXML) {
-    if (isUserSpecifiedFilesToInsert) {
-        // in case if someone did : [node app.js --files] and runs the program
-        if (filesToInsert.length) {
-            insertCsvIntoMysql(filesToInsert, "./", insertingCase);
-        } else {
-            console.error(new Error('No CSV or TXT files specified'));
-        }
+new Promise((resolve, reject) => {
+    if (filesToInsert.length && isUserSpecifiedFilesToInsert) {
+        resolve(filesToInsert)
     } else {
-        fs.readdir('./files/', (error, files) => {
-            if (error) {
-                console.error(error);
-                return;
-            }
-            let csvFiles = files.filter(file => /(\.csv)$/.test(file) || /(\.txt)$/.test(file));
-            if (!csvFiles.length) {
-                console.error(new Error('No CSV or TXT files found'));
-                return;
-            }
-
-            insertCsvIntoMysql(csvFiles, "./files/", insertingCase);
-
+        resolve(reaDirectory('./files/').then(files => files.map(file => `./files/${file}`)))
+            .catch(error => { reject(error) })
+    }
+}).then(files => {
+
+    return Promise.all(files.map(filePath => {

```

Comment:
This Promise.all() which then gets fed into .catch(console.error) means that each file is not treated individually. Essentially, all operations are "fired off" and if any of them fail then only the first one to fail will be output to the console. The rest will continue operating until they either finish or fail, and if they fail, then they will fail silently.

The best pattern is to take the .catch(console.error) and use it on the promise returned by the readAndParse call (and also get rid of Promise.all). This way, each operation that fails will output its error.